src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -181,13 +181,26 @@ def __init__(self, username, password, name=None, email=None, roles=None,
             self.refreshLastUpdate()
         else:
             self.lastUpdate = lastUpdate
+        self.needChangePassword = needChangePassword
 
     def refreshLastUpdate(self):
         self.lastUpdate = int(time.time())
 
     def set_password(self, password):
+        if bcrypt.checkpw(password.encode('utf8'), self.password.encode('utf8')):
+            raise DashboardException(msg='New password is the same as the current one.',
+                                 code='same-password',
+                                 component='Update User')
         self.password = password_hash(password)
         self.refreshLastUpdate()
+        if self.needChangePassword:
+            self.update_needChangePassword()
+            
+    def update_needChangePassword(self):
+        self.needChangePassword = not self.needChangePassword
+        self.refreshLastUpdate()
+
+    def set_roles(self, roles):
```

Comment:
This is not syntactically correct in Python. A void function needs at least an statement (`pass`).
```suggestion
    def set_roles(self, roles): pass
```

However, that will silently fail, so it's usually better to trigger an Exception like `NotImplementedError`.

Reply:
I will remove it, this declaration was copied by mistake.